### ISSUE TITLE
Error when installing MLRun using Helm

### ISSUE THREAD

**Issue by AliMirlou** [mlrun/mlrun#768]
Hi,
I tried to install MLRun by running the following commands:
``` bash
helm --namespace mlrun install mlrun-kit --wait --set global.registry.url=<some custom Docker registry URL> --set global.registry.secretName=registry-credentials v3io-stable/mlrun-kit
```
But got this error:
```
Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: [ValidationError(CustomResourceDefinition.spec): unknown field "subresources" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionSpec, ValidationError(CustomResourceDefinition.spec): unknown field "validation" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionSpec]
```
I searched for this error and it seems that "subresources" and "validation" fields were removed in v1 CustomResourceDefinition.

So I dumped the template and validated it manually:
``` bash
helm --namespace mlrun template mlrun-kit --wait --set global.registry.url=<some custom Docker registry URL> --set global.registry.secretName=registry-credentials v3io-stable/mlrun-kit > mlrun.yaml
kubectl apply --dry-run=client -f mlrun.yaml
```
But there were no errors!
I also searched the template for a v1 CustomResourceDefinition instead of v1beta1 but there were none!

Can anyone help?

---

Kubernetes version: 1.20.4
Helm version: 3.5.2

**Comment by Hedingber** [comment]
Hi @AliMirlou - thanks for reporting the issue.
Just to explain what's going on - this invalid CRD is coming from [the mpi-operator helm chart](https://github.com/v3io/helm-charts/tree/development/stable/mpi-operator) which as you can see [here](https://github.com/v3io/helm-charts/blob/development/stable/mlrun-kit/requirements.yaml#L11) is a requirement chart of the mlrun-kit helm chart
As you can see [here](https://github.com/v3io/helm-charts/blob/development/stable/mpi-operator/templates/mpijob-crd.yaml#L2) and [here](https://github.com/v3io/helm-charts/blob/development/stable/mpi-operator/templates/_helpers.tpl#L48) this chart "template" a different version for the crd based on your k8s version.
I guess that when you `helm template` it doesn't consider the server version, or just consider something else than in `helm install` and this is the reason you couldn't find template with `v1`
That also explain why you could deploy it through `kubectl apply` without errors - in what you deployed the version was v1beta1
I have successfully reproduced your issue and working on a fix

**Comment by AliMirlou** [comment]
@Hedingber Thanks.
But the thing about `helm install` having a different behavior from `helm template` doesn't make sense though! How is this an expected behavior? Or is it a bug in Helm?

**Comment by AliMirlou** [comment]
I also checked Helm documents and the object `.Capabilities.KubeVersion.GitVersion` was in Helm version 2 but not version 3. Might that be the problem? My guess is that this object was planned to be removed in version 3 and `helm template` has no access to it but `helm install` does (which is definitely a bug then) and that's causing the different behaviors between the two commands.

**Comment by Hedingber** [comment]
@AliMirlou 
Looking on their code - `GitVersion` is [still there](https://github.com/helm/helm/blob/master/pkg/chartutil/capabilities.go#L74) - indeed deprecated.
Looking on the references to the `getCapabilities` [function](https://github.com/helm/helm/blob/master/pkg/action/action.go#L239) it's being called only from the actions `install` `uninstall` and `upgrade`
So yeah, as we said, when doing `helm template` this does not get evaluated

**Comment by Hedingber** [comment]
Hi @AliMirlou 
I'm glad to tell you that [the fix](https://github.com/v3io/helm-charts/pull/561) to the mpi-operator chart was merged and released (version 0.4.1)
And the PR to bump the mpi-operator chart version in the mlrun-kit chart - https://github.com/v3io/helm-charts/pull/561 was also merged and released.
If you'll run `helm repo update` you should get the new mlrun-kit version - 0.1.3 - which should work for you 🎉 
Waiting for your feedback before closing the issue

**Comment by AliMirlou** [comment]
Hello again @Hedingber 
Thanks for the quick fix. I tested it and it worked fine. 👍
